src/components/forms/validator.ts

```diff
@@ -63,19 +63,26 @@ export const maxValue = (max: number | string) => (value: string): ValidatorRetu
 
 export const ok = (): undefined => undefined
 
-export const mustBeEthereumAddress = memoize(
+export const mustBeAddressHash = memoize(
   (address: string): ValidatorReturnType => {
+    const errorMessage = 'Must be a valid address'
     const startsWith0x = address?.startsWith('0x')
     const isAddress = getWeb3().utils.isAddress(address)
-
-    const errorMessage = `Must be a valid address${
-      isFeatureEnabled(FEATURES.DOMAIN_LOOKUP) ? ', ENS or Unstoppable domain' : ''
-    }`
-
     return startsWith0x && isAddress ? undefined : errorMessage
   },
 )
 
+export const mustBeEthereumAddress = memoize(
+  (address: string): ValidatorReturnType => {
+    const errorMessage = 'Must be a valid address, ENS or Unstoppable domain'
+    const result = mustBeAddressHash(address)
+    if (result !== undefined && isFeatureEnabled(FEATURES.DOMAIN_LOOKUP)) {
+      return errorMessage
+    }
+    return result
+  },
+)
+
```

Comment:
I have some doubt's regarding this function.

The idea is that if DOMAIN_LOOKUP is compatible with the network should add the domain part. But if it doesn't it still should show that the address is not valid

Reply:
It does, it returns the original error from `mustBeAddressHash` if the domain lookup isn't enabled.